test/unit/commands.test.ts

```diff
@@ -45,9 +51,12 @@ describe('BinMsg BSON utf8 validation', () => {
     // this is a sanity check to make sure nothing unexpected is happening in the deserialize method itself
 
     const options = { validation: { utf8: { writeErrors: false } as const } };
-    expect(BSON.deserialize(invalidUtf8ErrorMsgDeserializeInput, options)).to.deep.equals(
-      invalidUtf8InWriteErrorsJSON
-    );
+    const deserializerCall = () => BSON.deserialize(invalidUtf8ErrorMsgDeserializeInput, options);
```

Comment:
This comment applies to all tests in this file.

I think we should break apart the each test into two separate tests - one when bson ext is installed and one when it is not.  Right now, in the case that bson ext is installed, we're not actually testing what the test description says.  A test structure like the following would be easier to understand and would accurately describe the behavior of the test

```typescript
context('when bson ext is installed', function() {
  it('....', {})
})

context('when bson ext is not installed', function() {
  it('bson correctly deserializes data with replacement characters for invalid utf8 in writeErrors object', () => {
    ... 
  });
})

```

Reply:
This might require a larger restructure of the file but I think the better test organization is worth it

Reply:
Made two blocks, skipped stuff, and renamed tests